server/auth.go

```diff
@@ -628,12 +628,20 @@ func (s *Server) processClientOrLeafAuthentication(c *client, opts *Options) boo
 		}
 		// Hold onto the user's public key.
 		c.pubKey = juc.Subject
+		c.tags = juc.Tags
```

Comment:
I wonder if all those 3 should not be protected with c.mu.Lock().

Reply:
fixed